site/src/mosaic-plugins/ChildrenDocPaginatorPlugin.mjs

```diff
@@ -0,0 +1,63 @@
+const COMPONENT_OVERVIEW = {
+  title: "Overview",
+  route: "/salt/components/index",
+};
+
+function buildRouteNavigationData(pages) {
+  const allComponentIndexPages = pages
+    .filter((p) => new RegExp("components/[\\w-]+/index").test(p.route))
+    .sort((a, b) => a.route.localeCompare(b.route));
+
+  const pageIndexToNavigation = {};
+  for (let index = 0; index < allComponentIndexPages.length; index++) {
+    const page = allComponentIndexPages[index];
+    const prev =
+      index === 0
+        ? COMPONENT_OVERVIEW
+        : {
+            title: allComponentIndexPages[index - 1].title,
+            route: allComponentIndexPages[index - 1].route,
+          };
+
+    const next =
+      index === allComponentIndexPages.length - 1
+        ? undefined
+        : {
+            title: allComponentIndexPages[index + 1].title,
+            route: allComponentIndexPages[index + 1].route,
+          };
+
+    pageIndexToNavigation[page.route] = {
+      prev,
+      ...(next ? { next } : {}), // Don't want undefined or Mosaic will throw
+    };
+  }
+  return pageIndexToNavigation;
+}
+
+const ChildrenDocPaginatorPlugin = {
```

Comment:
I guess you can make this name more specific as its a plugin for the salt site...

Reply:
Updated